### DISCUSSION TITLE
Active profile name from VS code Extension API and create profile from extension API

### DISCUSSION THREAD

**Discussion by alagiribuddhar** [microsoft/vscode-discussions#809] [Extension Development QnA] [answered]
I am having use case where i need to check if particular profile exists and create one if not exits via Extension API. Is there way to do it in activate of Extension?

**Comment by alagiribuddhar** [comment]
@sandy081 Please provide any suggestions?

**Comment by alefragnani** [comment, marked as answer]
Hi @alagiribuddhar ,

Right now, there is no API available (nor proposals) to work with profiles. So, you won't be able to do check or create profiles.

I would suggest you to open a feature request in VS Code extension's repo (https://github.com/microsoft/vscode/issues/new?assignees=&labels=&projects=&template=feature_request.md), asking for such API, providing more details about your scenario, so the team could properly evaluate your request.

Hope this helps

**Reply by alagiribuddhar** [reply]
Thank you for suggestion.Will create feature request in vs code issues.

**Reply by cartu38** [reply]
Hi,
I've exactly same interest so request. Is feature request done already ? If yes please could you point me to.
Br,

**Reply by alagiribuddhar** [reply]
Hi,
The feature request can be found at https://github.com/microsoft/vscode/issues/192852
